presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -357,7 +357,9 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
             if (leftSource != node.getLeft() ||
                     rightSource != node.getRight() ||
                     !expressionEquivalence.areExpressionsEquivalent(session, newJoinPredicate, joinPredicate, types)) {
-                if (newJoinPredicate.equals(BooleanLiteral.TRUE_LITERAL)) {
+                if (node.getType() == JoinNode.Type.INNER && newJoinPredicate.equals(BooleanLiteral.TRUE_LITERAL)) {
```

Comment:
We should drive the code through here if the join predicate is TRUE, but we should retain the node.getType() for the execution so it knows what to do with it